plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -804,14 +854,25 @@ function OPDSBrowser:showDownloads(item)
             end
         end
     end
-    if (#type_buttons % 2 == 1) then -- we need even number of type buttons
-        table.insert(type_buttons, {text = ""})
+    -- handles adding download button(s)
+    if (download_buttons ~= {} and #download_buttons > 0) then
+        if (#download_buttons % 2 == 1 and #download_buttons > 1) then
+            table.insert(download_buttons, {text = ""})
+            for i = 2, #download_buttons, 2 do
+                table.insert(buttons, {download_buttons[i - 1], download_buttons[i]}) -- download buttons, two in a row
+            end
+        else
+            -- need the else to handle special case where we have one download button,
+            --   and it should span the width of the window
+            table.insert(buttons, download_buttons)
+        end
+        table.insert(buttons, {})  -- add separator for stream buttons
     end
-    for i = 2, #type_buttons, 2 do
-        table.insert(buttons, {type_buttons[i - 1], type_buttons[i]}) -- type buttons, two in a row
+    if (stream_buttons ~= {}) then
```

Comment:
That doesn't do what you think it does ;).

`{}` creates a *new* table object. For table objects, comparisons are done on their reference (i.e., an address), so for a match to, well, match, the reference has to be *stable*. This will *never* be the case here.

```
> print({})
table: 0x7fe622be2218
> print({})
table: 0x7fe622be2428
> print({})
table: 0x7fe622be2638
> print({})
table: 0x7fe622be2848
> print({})
table: 0x7fe622be2a58
> print({})
table: 0x7fe622be2c68
> print({})
table: 0x7fe622be2e78
> print({})
table: 0x7fe622be3088
> print({})
table: 0x7fe622be3298
```

Reply:
To check if a table is empty, if it's an array, you can either check the table's size (`#table`) or do a nil check on the first element; if it's a hash, you can see if `next` returns anything.

Reply:
So it should look for `#stream_buttons > 0` instead, right? Comparing number of entries in table rather than table address?

Reply:
Assuming it can't be `nil` (which it looks like it can't), yes ;).

Reply:
Should be fixed in the latest, took out parentheses and changed the logic so that it checks if it's not nil and that there is at least one entry in the table.